src/adagrams.js

```diff
@@ -1,8 +1,169 @@
-const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
-  },
+class Adagrams {
+  static frequencies() {
+    return [
+    ["A",9],
+    ["B", 2],
+    ["C", 2],
+    ["D", 4],
+    ["E", 12],
+    ["F", 2],
+    ["G", 3],
+    ["H", 2],
+    ["I", 9],
+    ["J", 1],
+    ["K", 1],
+    ["L", 4],
+    ["M", 2],
+    ["N", 6],
+    ["O", 8],
+    ["P", 2],
+    ["Q", 1],
+    ["R", 6],
+    ["S", 4],
+    ["T", 6],
+    ["U", 4],
+    ["V", 2],
+    ["W", 2],
+    ["X", 1],
+    ["Y", 2],
+    ["Z", 1],
+  ];}
+
+  static makeNewBag() {
+    let bag = [];
+    const freqs = Adagrams.frequencies();
+
+    for (let i=0; i<freqs.length; i++) {
+
+      let entry = freqs[i];
+
+      for (let j=0; j<entry[1]; j++) {
+        bag.push(entry[0]); 
+      }
+    }
+    return bag;
+  }
+  
+  static drawLetters() {
+    // draw 10 letters from newBag, return as array of 10 strings
+    let newBag = this.makeNewBag();
+    let tray = [];
+    // console.log(newBag);
+
+    for (let i=0; i < 10; i++) {
+      let randomIndex = Math.floor(Math.random() * newBag.length);
+
+      // remove from bag...
+      newBag.splice(randomIndex, 1);
+
+      // ...add to player's tray
+      tray.push(newBag[randomIndex]);
+    }
+    return tray;
+  }
+
+  static remove(element, array) {
+    // removes a single element from array
+    let garbageIndex = array.indexOf(element);
+    array.splice(garbageIndex, 1);
+  }
+
+  static usesAvailableLetters(inputAnyCase, lettersInHand) {
+    // input = string, supposedly made from the lettersInHand
+    // lettersInHand = array of 10 single-letter strings
+    // returns T if input is legit, else F
+    let input = inputAnyCase.toUpperCase();
+
+    for (let i=0; i<input.length; i++) {
+      if (lettersInHand.includes(input[i])) {
+        this.remove(input[i], lettersInHand);
+      } else {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  static chart() {
+    return[
+    [['A','E','I','O','U','L','N','R','S','T'], 1],
+    [['D','G'], 2],
+    [['B','C','M','P'], 3],
+    [['F','H','V','W','Y'], 4],
+    [['K'], 5],
+    [['J','X'], 8],
+    [['Q','Z'], 10],]; 
```

Comment:
Your data structure for this requires me to know what each nested array is supposed to look like at some index... which requires a lot of knowledge or explanation. I think this is fine for right now, but I (personally) probably prefer a more explicit object/name-value structure for that.